### ISSUE TITLE
Files order is not preserved in buffer mode

### ISSUE THREAD

**Issue by nfroidure** [gulpjs/gulp#300]
When the buffer mode is used and some plugins are doing some async stuffs on buffers, it leads to unordered files in the stream.

Creating a plugin that works asynchronously on buffers and preserve the inputed files order is over complicated.

The problem doesn't appear for streaming mode since all pipes are done immediatly.

**Comment by phated** [comment]
@contra is this still a thing?

**Comment by contra** [comment]
@phated yeah, its a fundamental issue with async transform streams

**Comment by phated** [comment]
Removed the bug and enhancement tags, as this is mainly a documentation issue.  Also, added help wanted tag if someone wants to write this up.

**Comment by piyushdubey** [comment]
I am new to the community and wish to contribute. I can work on the documentation.

**Comment by phated** [comment]
@piyushdubey I don't have any specific advice but generally, just submit doc changes as PRs and we can discuss back-and-forth.

**Comment by piyushdubey** [comment]
@phated Great! Should the docs be added under FAQs?

**Comment by denieler** [comment]
yes, if it is known issue, why it is not documented in `restore` function ?

**Comment by nfroidure** [comment]
@denieler gulp-filter is not part of gulp itself ;)

**Comment by denieler** [comment]
@nfroidure I thought it is opened issue in gulp-filter repo
